Pytest unit tests

webthethird opened the pull request:
Begins to implement unit tests in the CI pipeline, using Pytest.

webthethird committed:
Initial package structure for unit tests

webthethird committed:
Initial file structure for unit tests

webthethird committed:
Start test_helpers.py

webthethird committed:
Overhaul `helpers.get_pragma_versions_from_file`

webthethird committed:
Unit test for `get_pragma_versions_from_file`

webthethird committed:
Fill in remaining helper tests to do later

webthethird committed:
Begin test_code_generation.py

webthethird committed:
Add test data for code generation tests

webthethird committed:
Add initial Pytest CI script

webthethird committed:
Don't think I need to setup Node

webthethird committed:
Use Python 3.10

webthethird committed:
Update test dependencies in setup.py

webthethird committed:
Typo

webthethird committed:
Fix unit_test_runner.sh

webthethird committed:
Mypy and black

webthethird committed:
Add upload-coverage action

webthethird committed:
Fix Windows bug in `get_compilation_unit_name`

webthethird committed:
Rely on good-old os.path.join

webthethird committed:
Deal w/ Windows paths in `get_pragma_versions_from_file`

webthethird committed:
Fix typo

webthethird committed:
Add "dev" back to setup.py

0xalpharush commented on .github/workflows/pytest.yml:
```diff
@@ -0,0 +1,88 @@
+        # Only run coverage on ubuntu-latest.
+        run: |
+          if [ ${{ matrix.os }} = "ubuntu-latest" ]; then
+            TEST_ARGS="--cov=slither --cov-append"
```
```suggestion
            TEST_ARGS="--cov=diffusc --cov-append"
```

webthethird committed:
Caught in the act of copying from Slither

webthethird committed:
Black

webthethird committed:
Clean out unused contracts

webthethird committed:
Rename mode

webthethird committed:
Switch solc version in `test_compilation_unit_name`

webthethird committed:
Add test contracts for code generation

webthethird committed:
Add WIP `test_generate_contract_path_mode`

webthethird committed:
Add expected code and compare in test

webthethird committed:
Update code generation test files

webthethird committed:
Fix path bug on Windows

webthethird committed:
Fix auto-generated import comparison on Windows

webthethird committed:
Revert path separator replacement in comparison

webthethird committed:
Replace os path separator in import statements

webthethird committed:
Add `test_args_and_returns`

webthethird committed:
Add `function complexTypes` to CodeGeneration.sol

webthethird committed:
Add `function complexTypes` to CodeGeneration.sol

webthethird committed:
Use canonical name for structs in `get_solidity_function_returns`

webthethird committed:
Update expected interface code

webthethird committed:
Add `test_generate_contract_fork_mode`

webthethird committed:
Add `dotenv` to testing requirements

webthethird committed:
Add environment vars to pytest CI

webthethird committed:
Fix environment vars in pytest CI

webthethird committed:
Black

webthethird committed:
Black

webthethird committed:
Add `test_generate_config`

webthethird committed:
Add `test_generate_config`

webthethird committed:
Trying to set env vars in CI

webthethird committed:
Trying to set env vars in CI

webthethird committed:
Trying to set env vars in CI

webthethird committed:
Fix dotenv dependency

webthethird committed:
Fix env vars in pytest.yml
there may be a better way though

webthethird committed:
Remove dotenv dependency

webthethird committed:
Update solc version in tests to 0.8.11

webthethird committed:
Update and add more expected output files

webthethird committed:
Update and add more expected output files

webthethird committed:
Sort lists to avoid non-deterministic declaration order

webthethird committed:
Use targets' implementation interface, not proxy's,
if implementation address is specified when getting contract data

webthethird committed:
Add fifth test to each `test_generate_contract` test

webthethird committed:
Update expected code

webthethird committed:
Avoid comparing `corpusDir` in config in tests
because of tests running of Windows and Ubuntu
If the empty string is given as an argument to `generate_config_file`,
do not include `corpusDir` in config file

webthethird committed:
Black

webthethird committed:
Remove test_contract_data.py

webthethird committed:
Pylint

webthethird committed:
Retry getting Slither object from address 5 times

webthethird committed:
Black

webthethird committed:
Update test data for more coverage

webthethird committed:
Use relative path for tainted external contract imports

webthethird committed:
Use relative path for tainted external contract imports

webthethird committed:
Black

webthethird committed:
Fix expected file path in `test_contract_data_from_slither`
